### ISSUE TITLE
Python 3.10 syntax in 3.7 project?

### ISSUE THREAD

**Issue by raneq** [sbidoul/pyproject-dependencies#2]
Hi @sbidoul , thanks a lot for `whool` , I got here linked from that project.
I was also struggling with using current python build tools with the odoo-setuptools, and was stuck in there.

However, when trying `pyproject-dependencies`,  it crashes with this error:

``` 
$ pyproject-dependencies odoo/addons/sinli/
Traceback (most recent call last):
  File "/home/user/.pyenv/versions/devcontrol_modules/bin/pyproject-dependencies", line 5, in <module>
    from pyproject_dependencies.__main__ import main
  File "/home/user/.pyenv/versions/devcontrol_modules/lib/python3.7/site-packages/pyproject_dependencies/__main__.py", line 20, in <module>
    extra_environ: Mapping[str, str] | None = None,
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

```

In this other python project, they say the pipe syntax `|` is a 3.10 feature https://github.com/tiangolo/typer/issues/371#issuecomment-1073649190

Does this make sense to you? Would you consider adapting it to 3.8 syntax?
Thank you!

**Comment by sbidoul** [comment]
Sure. Would you like to do a PR to replace that with Union?

**Comment by raneq** [comment]
Ok I'll try!

**Comment by sbidoul** [comment]
Resolved in #3
